pkg/registry/apis/provisioning/jobs/worker.go

```diff
@@ -200,6 +148,88 @@ func (g *JobWorker) Process(ctx context.Context, job provisioning.Job, progress
 	default:
 		return nil, fmt.Errorf("unknown job action: %s", job.Spec.Action)
 	}
+}
+
+func (g *JobWorker) doSync(ctx context.Context,
+	repo repository.Repository,
+	job provisioning.Job,
+	parser *resources.Parser,
+	_ func(provisioning.JobStatus) error,
+) (*provisioning.JobStatus, error) {
+	var err error
+	cfg := repo.Config()
+
+	status := job.Status.ToSyncStatus(job.Name)
+	patch, err := json.Marshal(map[string]any{
+		"status": map[string]any{
+			"sync": status,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("SEND PATCH (starting): %s\n", (patch))
+	cfg, err = g.client.Repositories(cfg.Namespace).
+		Patch(ctx, cfg.Name, types.MergePatchType, patch, v1.PatchOptions{}, "status")
+	if err != nil {
+		logger.Warn("unable to update repo with job status", "err", err)
+	}
+
+	syncer, err := NewSyncer(repo, g.lister, parser)
+	if err != nil {
+		return nil, fmt.Errorf("error creating replicator")
+	}
+
+	var complete bool
+	if job.Spec.Sync != nil && job.Spec.Sync.Complete {
+		complete = job.Spec.Sync.Complete
+	}
+
+	// Sync the repository
+	ref, syncError := syncer.Sync(ctx, complete)
+	status = job.Status.ToSyncStatus(job.Name)
+	status.Hash = ref
+	status.Finished = time.Now().UnixMilli()
+	if syncError != nil {
+		status.State = provisioning.JobStateError
+		status.Message = []string{
+			"error running sync",
+			syncError.Error(),
+		}
+	} else {
+		status.State = provisioning.JobStateSuccess
+	}
+
+	// Update the resource stats
+	stats, err := g.lister.Stats(ctx, cfg.Namespace, cfg.Name)
+	if err != nil {
+		logger.Warn("unable to read stats", "error", err)
+	}
+	if stats == nil {
+		stats = &provisioning.ResourceStats{}
+	}
+
+	patch, err = json.Marshal(map[string]any{
+		"status": map[string]any{
+			"sync":  status,
+			"stats": stats.Items,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("SEND PATCH (after): %s\n", (patch))
+	_, err = g.client.Repositories(cfg.Namespace).
+		Patch(ctx, cfg.Name, types.MergePatchType, patch, v1.PatchOptions{}, "status")
```

Comment:
^^ using PATCH rather than UpdateStatus because this only updates *part* of the status, and the RV may change from the first time